de.zray.Facetracker.json

```diff
@@ -0,0 +1,64 @@
+{
+  "app-id": "de.zray.Facetracker",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "46",
+  "sdk": "org.gnome.Sdk",
+  "command": "facetracker-gui",
+  "finish-args": [
+    "--share=network",
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=all",
+    "--socket=wayland"
+  ],
+  "modules": [
+    {
+      "name": "facetracker-addional-files",
+      "buildsystem": "simple",
+      "build-commands": [
+        "mkdir -p /app/share/applications",
+        "install -D de.zray.Facetracker.desktop /app/share/applications/",
+
+        "mkdir -p /app/share/icons/hicolor/scalable/apps",
+        "install -D facetracker/data/icons/de.zray.Facetracker.svg /app/share/icons/hicolor/scalable/apps",
+
+        "install -Dp -m 644 de.zray.Facetracker.metainfo.xml /app/share/metainfo/de.zray.Facetracker.metainfo.xml"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/Z-Ray-Entertainment/Facetracker",
+          "tag": "24.7a3"
+        }
+      ]
+    },
+    {
+      "name": "OpenSeeFace",
+      "buildsystem": "simple",
+      "build-commands": [
+        "mkdir -p /app/share/facetracker/osf/",
+        "cp -r ./ /app/share/facetracker/osf/",
+        "chmod +x /app/share/facetracker/osf/facetracker"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/Z-Ray-Entertainment/Facetracker/releases/download/24.7a3/osf_pre_build.zip",
```

Comment:
why not building from source?

Reply:
To answer this question:

OpenSeeFace relies on the onnxruntime for which the official flatpak-pip-genertator can not generate the required requirements.json:

`Exception: Failed to get onnxruntime-1.18.1 source from https://pypi.org/pypi/onnxruntime/1.18.1/json`

The reason it fails is, that the flatpak-pip-generator does not allow for source archives in the whl format see:
https://github.com/flatpak/flatpak-builder-tools/blob/338ce9c6512d49d98ae9a508d219ffe19b5144e8/pip/flatpak-pip-generator#L86

Since the script fetches the source from pypi and pypi does not seem to contain a compatible format eg. bz2, gz, xz, zip the process of generating the json fails.

Manually adding the whl format to the array of supported archives causes other issues later down the line trying to build my entire project:
`ERROR: Could not find a version that satisfies the requirement onnxruntime~=1.18.1 (from versions: none)
ERROR: No matching distribution found for onnxruntime~=1.18.1`

As this is an issue with flatpak-builder and out of my control I decided to pre-build the binary for x86_64 and simply ship the pre-build via github.

This also answers the previous question as of why I did not allowed for aarch64 builds as I do not intend to support aarch64 at this point as this would require an additional pre-build just for aarch64. Also this would lead to the question if and how some could tell flathub to build the software differently for aarch64 as a different pre-build would be required than for x86.

Therefore I decided to cirumvnet all this issues by providing a pre-build for x86.  
Since I am neither a maintainer of OpenSeeFace and not the onnxruntime I also can not fix the root cause which is a missing source archive in a supported format.